### ISSUE TITLE
Cart randomly adding items during checkout process with registration

### ISSUE THREAD

**Issue by DanDvoracek** [opencart/opencart#8297]
**What version of OpenCart are you reporting this for?**
3.0.2.0

**Describe the bug**
Items in the cart get randomly doubled in the cart when users checkout and register an account in the process.

**To Reproduce**
1. Add 4 products to your cart
2. Then add 10 times another product
3. Then add 5 times another product again (5 and 10 are example, could be anything above 1)
4. Click on the cart widget's button "Order"
5. Go through checkout by registering a new account
6. Push the checkout to the end
7. Observe the cart and the checkout summary... You will see that at the last step, some products get randomly added to the cart.

* Unfortunately, you can't see that on the demo site as it is not fully usable/available for test...

**Expected behavior**
The cart should hold the amount of items the users order and not double them just before agreeing to pay...

**Screenshots / Screen recordings**

**Server / Test environment (please complete the following information):**
 - Local development? Deployed to a web server? --> all, even on a dev server
 - Operating system --> shared hosting
 - PHP version -> 7.2
 - Apache version --
 - Browser(s) tested with [e.g. chrome, safari - if applicable] Safari, Chrome, Firefox... Any really.

**Additional context**
I didn't alter the cart functions in any way. The only thing I found on the internet about it is the following thread: https://forum.opencart.com/viewtopic.php?t=211273

I tried the fix proposed by **letxobnav** but no luck. It doesn't change a thing actually.

**Comment by ADDCreative** [comment]
This looks too be a duplicate of https://github.com/opencart/opencart/issues/7375. The fix on the forum should work, although I haven't tested it myself. I did a workaround in a similar way. Probably need the refresh adding to some of the extensions that login as well. https://github.com/ADDCreative/opencart/commit/a8de6f3b6b8ba85771eb6738d37cba7b357a7ce2

**Comment by DanDvoracek** [comment]
Well the fix there doesn't work actually. Have you tried to repeat the bug before closing that issue?

**Comment by ADDCreative** [comment]
Did you click the refresh button to rebuild your modification cache after you made the changes? Did you also check that the files if in modification system/storage/modification contain the changes you made?

**Comment by DanDvoracek** [comment]
Ok I tried the second fix you've mentionned and correctly refreshed modifications. Now the cart gets empty right after registration succeeds. So it doesn't really fix my problem. Shall I remove this line in the Cart class constructor?

$this->db->query("DELETE FROM " . DB_PREFIX . "cart WHERE (api_id > '0' OR customer_id = '0') AND date_added < DATE_SUB(NOW(), INTERVAL 1 HOUR)");

**Comment by DanDvoracek** [comment]
Right I think I got it now. My latest comment did put me on track actually. I did try to apply **letxobnav** fix again from here (Thu Jun 04, 2020 8:25 am):

https://forum.opencart.com/viewtopic.php?t=211273

However, the cart was getting empty after users picked their shipping address. Therefore, I called the two methods (renew_session_id_old_items - transfer_items_to_new_customer) in the save function of the `ControllerCheckoutShippingAddress` class.

It seems to work just fine now. Will let you know if I spot anything else.
